### PR TITLE
Add missing container_timeout argument

### LINKED ISSUE #22986: "container_timeout" argument for providers.microsoft.azure.hooks.container_instance missing(?)
### Apache Airflow Provider(s)

microsoft-azure

### Versions of Apache Airflow Providers

apache-airflow-providers-microsoft-azure==3.7.2

apache-airflow-providers-amazon==3.2.0
apache-airflow-providers-celery==2.1.3
apache-airflow-providers-cncf-kubernetes==3.0.0
apache-airflow-providers-docker==2.5.2
apache-airflow-providers-elasticsearch==2.2.0
apache-airflow-providers-ftp==2.1.2
apache-airflow-providers-google==6.7.0
apache-airflow-providers-grpc==2.0.4
apache-airflow-providers-hashicorp==2.1.4
apache-airflow-providers-http==2.1.2
apache-airflow-providers-imap==2.2.3
apache-airflow-providers-mysql==2.2.3
apache-airflow-providers-odbc==2.0.4
apache-airflow-providers-postgres==4.1.0
apache-airflow-providers-redis==2.0.4
apache-airflow-providers-sendgrid==2.0.4
apache-airflow-providers-sftp==2.5.2
apache-airflow-providers-slack==4.2.3
apache-airflow-providers-sqlite==2.1.3
apache-airflow-providers-ssh==2.4.3

### Apache Airflow version

2.2.5 (latest released)

### Operating System

Server: Docker Desktop 4.7.0 (77141) & apache/airflow:2.2.5-python3.9

### Deployment

Docker-Compose

### Deployment details

_No response_

### What happened

I see from the [docs](https://github.com/apache/airflow/blob/main/airflow/providers/microsoft/azure/operators/container_instances.py) that the **AzureContainerInstancesOperator** accepts a "container_timeout" parameter (of type timedelta?).

When I supply the argument to the AzureContainerInstancesOperator my DAG serialization fails.

### What you think should happen instead

Either the argument should be removed from the docs:
- [Source1](https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/stable/_api/airflow/providers/microsoft/azure/operators/container_instances/index.html)
- [Source2](https://github.com/apache/airflow/blob/main/airflow/providers/microsoft/azure/operators/container_instances.py)

Or the argument should be re-introduced.

I'm hoping for the latter outcome as it would make the "cost control" kickback from my employer go over a lot smoother in our cloudification of an old Airflow instance.

### How to reproduce

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by spetrescu**
Fixes #22986
Added missing `container_timeout` argument in `providers.microsoft.azure.hooks.container_instance`

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
You need to fix tests. @spetrescu . I am planning to get new round of providers next week so it's good time to fix things and get it merged.

**Comment by spetrescu** [comment]
@potiuk please check now

**Review by uranusjr — APPROVED** [review]

File: airflow/providers/microsoft/azure/operators/container_instances.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import re
 from collections import namedtuple
 from time import sleep
+from datetime import timedelta
```
Comment (eladkal) [inline airflow/providers/microsoft/azure/operators/container_instances.py]:
```suggestion
from datetime import timedelta
from time import sleep
```

to fix static checks

**Comment by josh-fell** [comment]
I might be completely missing it, but where is this parameter being used? Seems like this change creates an instance attribute of `container_timeout` and remains unused.

**Comment by potiuk** [comment]
> I might be completely missing it, but where is this parameter being used? Seems like this change creates an instance attribute of `container_timeout` and remains unused.

It does sound suspicious. @spetrescu ?

**Pull request closed without merging**